### ISSUE TITLE
Restructure library structure

### ISSUE THREAD

**Issue by tobi-or-not-tobi** [SAP/spartacus#7959]
(TODO: tickets need updates)

The current library/code structure is based on a layers architecture, it contains (mainly) a core, components and style package.
This layered architecture is blocking 2 important things: 
- code-splitting (user experience)
- extensibility by other teams (developer experience)

The new package structure will use a feature based architecture. Features will be separated in various packages, where both core, component and styles are added to. We need to balance the number of packages vs the number of features. We like features to be organised in a few shared packages. Each feature ideally has it's own entry-point that can be used for code splitting. 

## Highlevel repo layout
[root]
 - storefrontapp (rename to app)
 - integration-libs
    - CDS (prep - Nik)
    - CPQ
    - CDC
    - vendor (let it die?)
        - Yotpo
  - features-libs
        - Checkout
        - Product
        - content
        - organization (b2b)
        - misc
            - storefinder
            - asm
            - qualtrics
  - core-libs (currently "projects")
    - core
    - component (storefrontlib)
        - layout
        - cms mapping
        - shared-components
    - assets
    - styles (storefrontstyles)
    - incubator 
    - setup (update / new)
        - Recipes
        - schematics 
- tools
    - e2e
    - dev-schematics
    - release stuff
    - scripts
    - stuff...

Next to these subfolder, there will be plenty of files, such as `angular.json`, 
With this setup, the current `project` folder is going to varnish. 

## Feature layout
A feature will contain all feature related bits and pieces, such as connectors, adapters, ngrx, facade, specific services, config, components, etc. A generic layout will be used as a template to all features, and we need to consistently use this template. However, it requires flexibility as not all features will have the same ingredients. 

The myaccount/company/costcenter is depicted as an example 

@Spartacus/organization
   - administration/ (Domain)
        - components/
        - core/
        - styles/
        - assets/
        - occ/
        - root/
        - testing/

Here's a screenshot of `my-account` lib.
<img width="328" alt="Screen Shot 2020-06-30 at 15 33 05" src="https://user-images.githubusercontent.com/11561723/86138237-84219e80-baee-11ea-9f14-8a5d9255fc9c.png">

## packages
- @spartacus/core 
- @Spartacus/components (current storefront)
- @Spartacus/Product
- @Spartacus/Checkout
- @Spartacus/misc (moving storefinder inside misc)
- @Spartacus/setup
- @Spartacus/organizzation

    
## Sub tasks

- [x] my account https://github.com/SAP/spartacus/issues/7960
- [ ] cds https://github.com/SAP/spartacus/issues/7961 (wait until CDS finishes the current ticket)
- [x] product https://github.com/SAP/spartacus/issues/7983

**Comment by WeizhengSap** [comment]
Do we also add `user` and `cart` features? And for `B2B`, it is not like other features (such as `product`, `checkout`). It is not an independent feature. Now, we added b2b checkout just inside our `checkout` module; and we don't have b2b `order details/history` (we added the b2b fields directly in `order details/history`). And, I think `multi-d` will be added into `product`. So, what we will put inside B2B feature lib? Maybe we can make `B2B` as an integration lib?
